llvmlite/binding/newpassmangers.py

```diff
@@ -0,0 +1,305 @@
+from ctypes import (c_bool, c_int)
+from llvmlite.binding import ffi
+from llvmlite.binding.initfini import llvm_version_info
+
+llvm_version_major = llvm_version_info[0]
+
+
+def create_new_module_pass_manager():
+    return NewModulePassManager()
+
+
+def create_new_function_pass_manager():
+    return NewFunctionPassManger()
+
+
+def create_pass_builder(tm, pto):
+    return PassBuilder(tm, pto)
+
+
+def create_pipeline_options():
+    return PipelineTuningOptions()
+
+
+class NewModulePassManager(ffi.ObjectRef):
+
+    def __init__(self, ptr=None):
+        if ptr is None:
+            ptr = ffi.lib.LLVMPY_CreateNewModulePassManager()
+        ffi.ObjectRef.__init__(self, ptr)
+
+    def run(self, module, pb):
+        ffi.lib.LLVMPY_NMPRun_module(self, pb, module)
+
+    def addVerifier(self):
+        ffi.lib.LLVMPY_AddVeriferPass(self)
+
+    def add_aa_eval_pass(self):
+        ffi.lib.LLVMPY_AddAAEvalPass_module(self)
+
+    def add_simplify_cfg_pass(self):
+        ffi.lib.LLVMPY_AddSimplifyCFGPass_module(self)
+
+    def add_loop_unroll_pass(self):
+        ffi.lib.LLVMPY_AddLoopUnrollPass_module(self)
+
+    def add_loop_rotate_pass(self):
+        ffi.lib.LLVMPY_LLVMAddLoopRotatePass_module(self)
+
+    def add_instruction_combine_pass(self):
+        ffi.lib.LLVMPY_LLVMAddInstructionCombinePass_module(self)
+
+    def add_jump_threading_pass(self, threshold=-1):
+        ffi.lib.LLVMPY_AddJumpThreadingPass_module(self, threshold)
+
+    # What will happen to the pointer to objects created with run() method
```

Comment:
What pointer is that? I can't see a pointer to objects created with the `run()` method.

Reply:
My bad, I was referring to the Analysis manager objects created in `LLVMPY_RunNew[Function|Module]PassManager` , but it's clear now they are just local objects with automatic storage, which should be managed implicitly.